edx_proctoring/views.py

```diff
@@ -286,6 +286,72 @@ def get(self, request, exam_id=None, course_id=None, content_id=None):  # pylint
         return Response(data)
 
 
+class StudentOnboardingStatusView(ProctoredAPIView):
+    """
+    Endpoint for the StudentOnboardingStatusView
+
+    Supports:
+        HTTP GET: returns the learner's onboarding status relative to the given course_id
+
+    HTTP GET
+        /edx_proctoring/v1/user_onboarding/status?course_id=edX/DemoX/Demo_Course
+
+    **Response Values**
+        * 'onboarding_status': string specifying the learner's onboarding status;
+            will return NULL if there are no onboarding attempts
+        * 'onboarding_link': link to the onboarding exam
+    """
+    def get(self, request):
+        """
+        HTTP GET handler. Returns the learner's onboarding status.
+        """
+        course_id = request.GET.get('course_id')
+
+        # If there are multiple practice exams, use the first exam
+        onboarding_exam = ProctoredExam.objects.filter(
+            course_id=course_id,
+            is_active=True,
+            is_practice_exam=True
+        ).first()
+        if not onboarding_exam:
+            return Response(
+                data='There is no onboarding exam related to this course id.',
+                status=404
+            )
+
+        # Get the most recent onboarding attempt + past attempts
+        recent_attempt = ProctoredExamStudentAttempt.objects.filter(
+            proctored_exam__course_id=course_id,
+            proctored_exam__is_practice_exam=True,
+            user__username=request.user.username,
+            taking_as_proctored=True,
+        ).first()
+        past_attempts = ProctoredExamStudentAttemptHistory.objects.filter(
+            proctored_exam__course_id=course_id,
+            proctored_exam__is_practice_exam=True,
+            user__username=request.user.username,
+            taking_as_proctored=True,
+        ).order_by('-started_at')
+
+        # Default to the most recent attempt if there are no verified attempts
+        relevant_attempt = recent_attempt
+        for attempt in past_attempts:
+            if attempt.status == ProctoredExamStudentAttemptStatus.verified:
```

Comment:
Can you double check in production that all onboarding exam with Software Secure have attempt status to be `verified`? If they don't have that status, which status do those have? I want to make sure this endpoint works the same in the case of Software Secure practice exams.

Reply:
It should be the same: https://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/proctored_exams/rpnow_results.html#values-in-the-status-column